com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaBuilder.java

```diff
@@ -184,37 +198,45 @@
     public void build(Task<Void> task) throws CompileExceptionError, IOException {
 
         LuaModule.Builder builder = LuaModule.newBuilder();
-        byte[] scriptBytes = task.input(0).getContent();
-        String script = new String(scriptBytes);
-        List<String> modules = LuaScanner.scan(script);
+        final byte[] originalScriptBytes = task.input(0).getContent();
+        final String originalScript = new String(originalScriptBytes, "UTF-8");
+
+        // run the Lua scanner to get and remove require and properties
+        LuaScanner scanner = new LuaScanner();
+        String script = scanner.parse(originalScript);
+        List<String> modules = scanner.getModules();
+        List<LuaScanner.Property> properties = scanner.getProperties();
 
+        // add detected modules to builder
         for (String module : modules) {
             String module_file = String.format("/%s.lua", module.replaceAll("\\.", "/"));
             BuilderUtil.checkResource(this.project, task.input(0), "module", module_file);
             builder.addModules(module);
             builder.addResources(module_file + "c");
         }
+
+        // add detected properties to builder
         Collection<String> propertyResources = new HashSet<String>();
-        List<LuaScanner.Property> properties = LuaScanner.scanProperties(script);
         PropertyDeclarations propertiesMsg = buildProperties(task.input(0), properties, propertyResources);
         builder.setProperties(propertiesMsg);
         builder.addAllPropertyResources(propertyResources);
-        LuaSource.Builder srcBuilder = LuaSource.newBuilder();
-        byte[] scriptBytesStripped = constructStrippedLuaCode(task.input(0).getContent());
-
-        /*
-        // For now it will always return, or throw an exception. This leaves the possibility of
-        // disabling bytecode generation.
-        byte[] bytecode = constructBytecode(task, task.input(0).getPath(), scriptBytesStripped);
-        if (bytecode != null)
-            srcBuilder.setBytecode(ByteString.copyFrom(bytecode));
-        */
 
-        srcBuilder.setFilename(task.input(0).getPath());
+        // create any builder plugins
+        if (luaBuilderPlugins == null) {
+            luaBuilderPlugins = PluginScanner.createPlugins("com.dynamo.bob.pipeline", LuaBuilderPlugin.class);
```

Comment:
Plural?

Reply:
Change to only accept and return a single plugin.